From f4b9adb95345ba1632c2fc25da7df39c855b324e Mon Sep 17 00:00:00 2001
From: Nikhil Jain C S <nikhil.jain.c.s@intel.com>
Date: Fri, 24 Jun 2022 16:54:25 +0530
Subject: [PATCH] Polarity Issue

Reversing the polarity on D-Bus

Tested on AVC.

Power is off -> OSStatus Inactive
Power is On -> OSStatus Standby

Signed-off-by: Nikhil Jain C S <nikhil.jain.c.s@intel.com>
Upstream-Status: Pending
---
 src/power_control.cpp | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/power_control.cpp b/src/power_control.cpp
index 05aa45c..9d5bd33 100644
--- a/src/power_control.cpp
+++ b/src/power_control.cpp
@@ -2237,12 +2237,12 @@ static void postCompleteHandler(bool state)
     if (!state)
     {
         sendPowerControlEvent(Event::postCompleteAssert);
-        setOperatingSystemState(OperatingSystemStateStage::Standby);
+        setOperatingSystemState(OperatingSystemStateStage::Inactive);
     }
     else
     {
         sendPowerControlEvent(Event::postCompleteDeAssert);
-        setOperatingSystemState(OperatingSystemStateStage::Inactive);
+        setOperatingSystemState(OperatingSystemStateStage::Standby);
     }
 }
 
@@ -3277,8 +3277,8 @@ int main(int argc, char* argv[])
     else
     {
         osState = getProperty(postCompleteConfig) > 0
-                      ? OperatingSystemStateStage::Inactive
-                      : OperatingSystemStateStage::Standby;
+                      ? OperatingSystemStateStage::Standby
+                      : OperatingSystemStateStage::Inactive;
     }
 
     osIface->register_property(
-- 
2.17.1

